File: FbxCppTests/FbxTimeSpanTest.cpp

#include "Tests.h"

using namespace std;

void FbxTimeSpan_Create_HasDefaults()
{
    // given:
    FbxManager* manager = FbxManager::Create();
    FbxTimeSpan ts;
    ts = FbxTimeSpan();

    // expect:
    AssertEqual(0LL, ts.GetStart().Get());
    AssertEqual(0LL, ts.GetStop().Get());
    AssertEqual(0LL, ts.GetDuration().Get());
    AssertEqual(0LL, ts.GetSignedDuration().Get());
    AssertEqual(1, ts.GetDirection());
}

void FbxTimeSpan_Create_WithArguments()
{
    // given:
    FbxManager* manager = FbxManager::Create();
    FbxTimeSpan ts;

    // when:
    ts = FbxTimeSpan(FbxTime(141120000L), FbxTime(423360000L));
    // then:
    AssertEqual(141120000LL, ts.GetStart().Get());
    AssertEqual(423360000LL, ts.GetStop().Get());
    AssertEqual(282240000LL, ts.GetDuration().Get());
    AssertEqual(282240000LL, ts.GetSignedDuration().Get());
    AssertEqual(1, ts.GetDirection());

    // when:
    ts = FbxTimeSpan(FbxTime(423360000L), FbxTime(141120000L));
    // then:
    AssertEqual(423360000LL, ts.GetStart().Get());
    AssertEqual(141120000LL, ts.GetStop().Get());
    AssertEqual(282240000LL, ts.GetDuration().Get());
    AssertEqual(-282240000LL, ts.GetSignedDuration().Get());
    AssertEqual(-1, ts.GetDirection());
}

void FbxTimeSpanTest::RegisterTestCases()
{
    AddTestCase(FbxTimeSpan_Create_HasDefaults);
    AddTestCase(FbxTimeSpan_Create_WithArguments);
}

File: FbxSharpTests/FbxTimeSpanTest.cs
using System;
using NUnit.Framework;
using FbxSharp;

namespace FbxSharpTests
{
    [TestFixture]
    public class FbxTimeSpanTest : TestBase
    {
        [Test]
        public void FbxTimeSpan_Create_HasDefaults()
        {
            // given:
            FbxTimeSpan ts;
            ts = new FbxTimeSpan();

            // expect:
            Assert.That(ts.GetStart().Get(), Is.EqualTo(0L));
            Assert.That(ts.GetStop().Get(), Is.EqualTo(0L));
            Assert.That(ts.GetDuration().Get(), Is.EqualTo(0L));
            Assert.That(ts.GetSignedDuration().Get(), Is.EqualTo(0L));
            Assert.That(ts.GetDirection(), Is.EqualTo(1));
        }

        [Test]
        public void FbxTimeSpan_Create_WithArguments()
        {
            // given:
            FbxTimeSpan ts;

            // when:
            ts = new FbxTimeSpan(new FbxTime(141120000L), new FbxTime(423360000L));
            // then:
            Assert.That(ts.GetStart().Get(), Is.EqualTo(141120000L));
            Assert.That(ts.GetStop().Get(), Is.EqualTo(423360000L));
            Assert.That(ts.GetDuration().Get(), Is.EqualTo(282240000L));
            Assert.That(ts.GetSignedDuration().Get(), Is.EqualTo(282240000L));
            Assert.That(ts.GetDirection(), Is.EqualTo(1));

            // when:
            ts = new FbxTimeSpan(new FbxTime(423360000L), new FbxTime(141120000L));
            // then:
            Assert.That(ts.GetStart().Get(), Is.EqualTo(423360000L));
            Assert.That(ts.GetStop().Get(), Is.EqualTo(141120000L));
            Assert.That(ts.GetDuration().Get(), Is.EqualTo(-282240000L));
            Assert.That(ts.GetSignedDuration().Get(), Is.EqualTo(-282240000L));
            Assert.That(ts.GetDirection(), Is.EqualTo(-1));
        }
    }
}
